tensorflow/compiler/xla/service/gpu/BUILD

```diff
@@ -1057,9 +1057,10 @@ cc_library(
         "//tensorflow/compiler/xla/service:buffer_assignment",
         "//tensorflow/compiler/xla:status",
         "//tensorflow/core:tflite_portable_logging",
-        "//tensorflow/stream_executor/cuda:cublas_plugin",
+        "//tensorflow/core/platform/default/build_config:cublas_plugin",
         "//tensorflow/stream_executor:device_memory",
         "//tensorflow/stream_executor:stream_header",
+        "//tensorflow/stream_executor/cuda:cublas_lt_header",
```

Comment:
Why only have the header here? Is it because otherwise the rule would be in both the shared libraries mentioned in the PR description?

Reply:
Thanks @reedwm for the review!

Yes, that's exactly it. The target `//tensorflow/stream_executor/cuda:cublas_plugin` is the "real" cublas which is currently built into `libtensorflow_framework.so` alongside all of the other cuda libraries.

Since this target here `cublas_lt_matmul_thunk` is built into `pywrap_tensorflow_internal.so`, if it depends on the real cublas `//tensorflow/stream_executor/cuda:cublas_plugin`, then it will cause a duplicate registration of cublas when both shared libraries, which looks like: `2022-06-28 23:43:46.827031: E tensorflow/stream_executor/cuda/cuda_blas.cc:3005] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been
 registered` (from [original issue](https://github.com/tensorflow/tensorflow/issues/56630)).

To fix it, I had to make sure all targets in `pywrap_tensorflow_internal.so` don't depend on the real cublas and only use the headers if they want to call cublas APIs. This pattern is followed by all other uses of cuda libraries such as cudnn, cufft, curand, etc. It seems whoever wrote the breaking commit did not know about this pattern and was using the monolithic build which is why they didn't see any errors.

Reply:
Thank you for the explanation! FYI, I had a similar problem when adding a global variable that required a [similar solution](https://github.com/tensorflow/tensorflow/blob/908465c098c0721466a2d8d156cfbf987dff85ad/tensorflow/core/util/BUILD#L654-L660). I didn't know about `check_deps` before, so I'll try using it .

Reply:
Thanks @reedwm!